meson.build

```diff
@@ -555,8 +555,11 @@ endif
 #####################################################################
 
 version_tag = get_option('version-tag')
-if version_tag != ''
+if version_tag != '' or get_option('mode') == 'developer'
         vcs_data = configuration_data()
+        if version_tag == ''
+                version_tag = 'v@0@-dev'.format(meson.project_version().to_int() + 1)
```

Comment:
`+ 1` for version here causes something strange after we release RC version, that is, after v248-rc1 is released, the tag becomes `v249-dev`. I think `+ 1` should be dropped.